src/main/kotlin/cacophony/controlflow/Variable.kt

```diff
@@ -0,0 +1,17 @@
+package cacophony.controlflow
+
+import cacophony.semantic.syntaxtree.Definition
+
+sealed class Variable
+
+class SourceVariable(
+    val definition: Definition,
+) : Variable()
+
+sealed class Register() : Variable()
+
+data object VirtualRegister : Register()
+
+data object FixedRegister : Register() {
+    // TODO
+}
```

Comment:
**question:** Why are these singleton objects and not classes?

Reply:
Right, they should be classes.